src/storage/calendar.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect } from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+type CalendarEntry = {
+  date: string;
+  start: string;
+  end: string;
+  text: string;
+  icon: string;
+  person: string;
+};
+type Calendar = CalendarEntry[];
+type ModifyCalendar = {
+  add: (entry: CalendarEntry) => boolean;
+};
+
+export const calendarKey: string = "calendar";
+export const calendarDefault: Calendar = [];
+
+/**
+ * Compares two calendar entries and returns `true` if they are equal.
+ *
+ * @param a - The first calendar entry
+ * @param b - The second calendar entry
+ * @returns a === b
+ */
+const entryEq = (a: CalendarEntry, b: CalendarEntry) => {
+  return a.date === b.date
+      && a.start === b.start
+      && a.end === b.end
+      && a.text === b.text
+      && a.icon === b.icon
+      && a.person === b.person;
+}
```

Comment:
a and b should not have to have an explanation. Use eg calenderEntry1 och calenderEntry2